### PR TITLE
Adding integration tests for multiple versions of OpenSearch

### PR DESCRIPTION
**Pull request by VachaShah**
Signed-off-by: Vacha Shah <vachshah@amazon.com>

### Description
Adding support for integration tests to run against all versions of OpenSearch. Also adding compatibility matrix for the client.
 
### Issues Resolved
[List any issues this PR will resolve]
 
### Check List
- [X] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: .ci/opensearch/docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,17 @@
+version: '3'
+
+services:
+
+  opensearch:
+    build:
+      context: .
```
Comment (saratvemulapalli) [inline .ci/opensearch/docker-compose.yml:7]: Curious what does this signify ?

File: .ci/opensearch/docker-compose.yml [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+      dockerfile: Dockerfile
+      args:
+        - SECURE_INTEGRATION=${SECURE_INTEGRATION:-true}
+        - OPENSEARCH_VERSION=${OPENSEARCH_VERSION}
+    environment:
+      - discovery.type=single-node
+      - bootstrap.memory_lock=true
+    ports:
+      - "9200:9200"
```
Comment (saratvemulapalli) [inline .ci/opensearch/docker-compose.yml:16]: We should also open up 9600 for [perf analyzer](https://github.com/opensearch-project/performance-analyzer)

**Review by saratvemulapalli — COMMENTED** [review]
Overall it looks good to me.
Couple of minor questions.

File: .ci/opensearch/docker-compose.yml (hunk 1 above)
Comment (VachaShah) [reply .ci/opensearch/docker-compose.yml:7]: The context is for a path to a directory containing a Dockerfile https://docs.docker.com/compose/compose-file/compose-file-v3/#build

File: .ci/opensearch/docker-compose.yml (hunk 2 above)
Comment (VachaShah) [reply .ci/opensearch/docker-compose.yml:16]: Currently clients don't support plugins, so the port added is only 9200 for OpenSearch.

File: .github/workflows/test-integration.yml [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+          - 1.0.1
+          - 1.1.0
+          - 1.2.0
+          - 1.2.1
+          - 1.2.2
+          - 1.2.3
+          - 1.2.4
+          - 1.3.0
+    steps:
+      - name: Set up JDK 14
```
Comment (dblock) [inline .github/workflows/test-integration.yml:27]: For a future PR, you should downgrade to JDK 11, or turn this into a matrix that does a combination of versions of OpenSearch and JDK, like [so](https://github.com/slack-ruby/slack-ruby-client/blob/master/.github/workflows/test.yml#L12).

**Review by dblock — APPROVED** [review]
Very nice. Let's get JDK sorted in a future PR.

**Review by saratvemulapalli — APPROVED** [review]

**Pull request merged**